Src/WindowsServer/WindowsServer.Shared/Implementation/AzureMetadataRequestor.cs

```diff
@@ -0,0 +1,120 @@
+        /// <returns>An array of field names available, or null.</returns>
+        public async Task<IEnumerable<string>> GetAzureInstanceMetadataComputeFields()
+        {
+            string allFieldsResponse = string.Empty;
+            string metadataRequestUrl = $"{baseImdsUrl}?{imdsTextFormat}&{imdsApiVersion}";
+
+            allFieldsResponse = await this.MakeAzureMetadataRequest(metadataRequestUrl);
+
+            string[] fields = allFieldsResponse?.Split(Environment.NewLine.ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
+
+            if (fields == null || fields.Length <= 0)
+            {
+                WindowsServerEventSource.Log.CannotObtainAzureInstanceMetadata();
+            }
+
+            return fields;
+        }
+
+        private async Task<string> MakeAzureMetadataRequest(string metadataRequestUrl)
+        {
+            string requestResult = string.Empty;
+
+            SdkInternalOperationsMonitor.Enter();
+            try
+            {
+#if NETSTANDARD1_3
+
+                using (var getFieldValueClient = new HttpClient(new HttpClientHandler() { MaxRequestContentBufferSize = MAX_IMS_RESPONSE_BUFFER_SIZE }))
+                {
+                    getFieldValueClient.DefaultRequestHeaders.Add("Metadata", "True");
+                    requestResult = await getFieldValueClient.GetStringAsync(metadataRequestUrl).ConfigureAwait(false);
+                }
+
+#elif NET45 || NET46
+
+                WebRequest request = WebRequest.Create(metadataRequestUrl);
+                request.Method = "GET";
+                request.Headers.Add("Metadata", "true");
+                using (WebResponse response = await request.GetResponseAsync().ConfigureAwait(false))
+                {
+                    if (response is HttpWebResponse)
+                    {
+                        var httpResponse = (HttpWebResponse)response;
+                        if (httpResponse.StatusCode == HttpStatusCode.OK)
+                        {
+                            char[] buffer = new char[maxImsResponseBufferSize];
+                            StreamReader content = new StreamReader(httpResponse.GetResponseStream());
+                            {
+                                int bufferIndex = await content.ReadAsync(buffer, 0, buffer.Length);
+
+                                // this will probably never exceed the buffer's size in bytes returned, scrap anything else that is left
+                                if (bufferIndex < buffer.Length - 1)
+                                {
+                                    requestResult = buffer.ToString();
+                                }
+                                else
+                                {
+                                    WindowsServerEventSource.Log.AzureInstanceMetadataRequestFailure(metadataRequestUrl, "Content received from Azure Metadata Instance service exceeds expected size, failing the call to avoid potential attack", string.Empty);
```

Comment:
these error logging is not done for netstandard

Reply:
No - what i meant is in the above code, you have an If {netstandard} {} block and if{net45} {} block. The error logging is not done in the netstandard block.